### ISSUE TITLE
Downloading github files through proxy

### ISSUE THREAD

**Issue by JohanLarsson** [fsprojects/Paket#1832]
### Description

Paket does not work with github files from behind proxy.

When building paket.sln:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Could not download from https://raw.githubusercontent.com/fsharp/FAKE/7f9fe8f546f6adec66febeec31e50821b60814d6/modules/Octokit/Octokit.fsx
 Message: One or more errors occurred.
 Details: The underlying connection was closed: An unexpected error occurred on a send.		C:\Git\Third Party\Paket\integrationtests\Paket.IntegrationTests\paket.references	0	
```
### Repro steps

```
#r "System.dll"
open System.Net

let https = @"https://raw.githubusercontent.com/fsharp/FAKE/f4024fd5b790485828c1cfc002190716eee97597/modules/Octokit/Octokit.fsx";
let http = @"http://stackoverflow.com/"

let download (url:string) =
    printfn "%s" url;
    try
      use client = new WebClient()
      // client.Proxy.Credentials <- System.Net.CredentialCache.DefaultNetworkCredentials
      client.DownloadString url |> ignore
      printfn "Success!";
    with
    | ex -> printfn "Exception! %s " (ex.Message);

download http;
download https;
```

Uncommenting the proxy line gives success for http but exception for https.

I don't know much about proxies, been testing a bunch of stackoverflow answers but nothing has worked. If anyone has suggestions please paste a modified script and I'll run it and output the results.

### Known workarounds

a) Download the files manually and turn off auto-restore.
b) [This](https://github.com/JohanLarsson/WpfGet/tree/master/WpfGet) is a nasty hack, but it works for downloading the file. It is Windows only and brings in WPF or WinForms references which is not ideal.

### Related information 

* Win 7
* Latest paket and every version I have tried.
* .NET
* Don't know much about the proxy in use.

**Comment by forki** [comment]
https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Utils.fs#L355 is how we handle proxies.

@JohanLarsson that one gets called during github download, right?

**Comment by vasily-kirichenko** [comment]
@forki I've never used GitHub deps.

**Comment by vilinski** [comment]
should be like here:

      use client = new WebClient()
      let proxy = WebRequest.DefaultWebProxy
      proxy.Credentials <- CredentialCache.DefaultCredentials
      client.Proxy <- proxy

**Comment by vasily-kirichenko** [comment]
@vilinski no, this does not work for me.

**Comment by vilinski** [comment]
That's strange, it should work, works for me behind authorized corp firewall. Other method  - not to touch code at any places, but set this into app.config:

      <system.net>
          <defaultProxy useDefaultCredentials="true" />
      </system.net>

**Comment by forki** [comment]
strange thing is all other http calls seem to work - @JohanLarsson are we really calling https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Utils.fs#L355 for github deps?

**Comment by JohanLarsson** [comment]
@forki I'll change to `client.Proxy <- getDefaultProxyFor url` in the script and test it, need to go to work to do it, hopefully tonight.
Regarding if it is called I'll try the manual download workaround and set a bp.
Reporting back as soon as I have results.

**Comment by JohanLarsson** [comment]
@forki Confirmed bp gets hit in `getDefaultProxyFor`when running `install`.

**Comment by forki** [comment]
Yeah but is that still failing? It could still be a normal nuget call (from
unrelated code)

**Comment by JohanLarsson** [comment]
It fails but the stack does not reveal much so I can't tell for sure yet if it got called in the failing path. 
I tried using a `paket.dependencies` with only a github ref but then I ended up [here](https://github.com/fsprojects/Paket/blob/master/src/Paket/Program.fs#L407)

**Comment by JohanLarsson** [comment]
Confirmed did not work:
https://github.com/fsprojects/Paket/issues/1832#issuecomment-235186799
